src/proposal.mligo

```diff
@@ -362,51 +374,36 @@ let handle_proposal_is_over
     in
     let cons = fun (l, e : operation list * operation) -> e :: l in
     let ops = List.fold cons ops new_ops in
-    let store = delete_proposal (start_date, proposal_key, store) in
-    (ops, store, counter)
-  else (ops, store, counter)
+    let store = delete_proposal (proposal.start_date, proposal_key, store) in
+    (ops, store)
+  else (ops, store)
 
-// Flush all proposals that passed their voting period.
-let flush(n, config, store : nat * config * storage): return =
+// Elem function for list of `proposal_key`
+[@inline]
+let elem_proposal_key_list (l, ls : proposal_key * proposal_key list) : bool =
+  let check_equal (acc, e : bool * proposal_key) =
+        if e = l then true else false
+  in List.fold check_equal ls false
+
+// Flush proposals that are provided in the parameter. Other proposals are not affected.
+let flush(keys, config, store : proposal_key list * config * storage): return =
   let store =
-    if n = 0n
+    if List.length keys = 0n
       then
         (failwith("BAD_ENTRYPOINT_PARAMETER") : storage)
       else store
```

Comment:
I think there is no need to fail here and we can just let `List.fold` handle the empty case as well.